lib/pescarte/modulo_pesquisa/relatorio_compiler.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Pescarte.RelatorioCompiler do
+  alias Pescarte.ModuloPesquisa.Repository
+  alias PescarteWeb.RelatorioHTML
+
+  def compilar_relatorios(relatorios_selecionados) do
+    relatorios = Enum.map(relatorios_selecionados, &find_relatorio/1)
+    htmls = Enum.map(relatorios, &RelatorioHTML.content/1)
+    pdfs = Enum.reduce(htmls, [], fn html, acc -> [{:html, html} | acc] end)
+
+    with {:ok, pdf_compilado} <- ChromicPDF.print_to_pdf(pdfs),
+         zip_file <- criar_zip_file(pdf_compilado) do
+      elem(zip_file, 1)
+    end
+  end
+
+  def gerar_pdf(relatorio_id) do
+    relatorio = find_relatorio(relatorio_id)
+
+    with {:ok, pdf} <-
+           [content: RelatorioHTML.content(relatorio), size: :a4]
+           |> ChromicPDF.Template.source_and_options()
+           |> ChromicPDF.print_to_pdf() do
+      Base.decode64(pdf)
+    end
```

Comment:
o relatório pode não existir por algum erro ou inconsistência do sistema, logo, é recomendável que verifiquemos isso antes de quebrar em runtime

```suggestion
    if relatorio = find_relatorio(relatorio_id) do
      with {:ok, pdf} <-
             [content: RelatorioHTML.content(relatorio), size: :a4]
             |> ChromicPDF.Template.source_and_options()
             |> ChromicPDF.print_to_pdf() do
        Base.decode64(pdf)
      end
    else
      {:error, :pdf_not_found}
    end
```